### ISSUE TITLE
Support High Contrast Text/Fonts

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#30974]
Android has an A11y setting for high contrast text/fonts.  Fuchsia would like this as well.

The framework and engine don't currently respect this flag.  We should implement it.

On Android, this is documented as forcing all fonts to black or white: https://support.google.com/accessibility/android/answer/6151855?hl=en (on my personal phone, this setting is actually labeled as `High contrast fonts`).

I do see that in some cases, where you would have white text on a white background, the setting forces a black stroke around the font:

![image](https://user-images.githubusercontent.com/8620741/56056617-92c93d80-5d11-11e9-8bbc-f9ef55383acd.png)

**Comment by dnfield** [comment]
On closer inspection, it _always_ adds that black stroke, even if the background is dark. The stroke just isn't as visible if the background is dark.

**Comment by dnfield** [comment]
It also adds a white stroke to black text.

**Comment by dnfield** [comment]
I'm not sure this can be implemented on Android - the interfaces for detecting it are not allowed for access on P/Q, and I can't seem to find a public alternative.

**Comment by dnfield** [comment]
Some things that we want to avoid:

1. Causing clipping of text or forcing a significantly changed layout if we have to change the overall pixel area of the text to render it in high contrast
2. Making sure text is sufficiently visible if we do end up shifting to a static color - e.g. the white text with a black stroke on a grey background is hard to read, especially if it's small.

An ideal scenario would be that we render the text based on the background color, which may not be practical.

**Comment by dnfield** [comment]
Action items on this one:

- We don't want to do this the way Android does it currently.  Android is working on changing that, and the feature has several known issues that need to be resolved.
- This has been deprioritized for Fuchsia for the time being.

**Comment by rekonvald** [comment]
Hi! I need support fro High Contrast Text/Fonts on android devices. When I change it through Accessibility settings nothing happens

**Comment by gnprice** [comment]
> 1. Causing clipping of text or forcing a significantly changed layout if we have to change the overall pixel area of the text to render it in high contrast

Android seems to successfully avoid changing the layout today; I tested on Android 13 and in the system settings app, Gmail, and a React Native app. When I turn the setting on and off, all the text keeps exactly the same layout. (I'm not sure if that was already true in 2019.)

That also means it doesn't cause any clipping of the text's strokes. There sometimes is clipping of the white or black border, though.


> 2. Making sure text is sufficiently visible if we do end up shifting to a static color - e.g. the white text with a black stroke on a grey background is hard to read, especially if it's small.

The current Android implementation does have this problem, though.



> * We don't want to do this the way Android does it currently.  Android is working on changing that, and the feature has several known issues that need to be resolved.

I'm curious whether those changes have happened, or if the known issues in Android's implementation as of 2019 are still unsolved.

**Comment by dnfield** [comment]
The last update on an issue I can see internally indicates this did not make it for Android T, but no updates on it about further versions.

AFAICT, the fundamental issue is no clear understanding of how to actually make the text higher contrast. If Android arrives at a way to do that, Flutter should be able to do something similar. But until they're happy with a solution, they're unlikely to open up the API to detect whether the feature is enabled or not, which is currently still marked as `@hide`.

**Comment by gnprice** [comment]
Makes sense. Thanks for the update!

**Comment by chunhtai** [comment]
cc @Renzo-Olivares since this is a typographic a11y feature, and this seems like something we should support

**Comment by gnprice** [comment]
Cool, thanks @danagbemava-nc for the update.

From the before/after screenshots at that second link:
![Image](https://github.com/user-attachments/assets/e0ad26dd-2e5c-4a80-92e4-8e692b69cea9)
it looks like they've changed the design: it's no longer an outline tightly hugging the strokes of the text, but instead a solid background pill drawn under the tex, covering roughly the whole bounding box of the text (but with a rounded shape). That does seem like it resolves the problems discussed above with the previous design.
